tests/neynarAPI.ts

```diff
@@ -0,0 +1,609 @@
+          expectDefined(reply);
+          expectDefined(publishedCast);
+          await sleep(1000);
+          await client.deleteCast(signerUuid, reply.hash);
+          await client.deleteCast(signerUuid, publishedCast.hash);
+        });
+
+        // no error thrown at the moment
+        it.skip("cannot delete a cast from a different author", async function () {
+          let errorThrownAndParsedCorrectly = false;
+          try {
+            await client.deleteCast(
+              signerUuid,
+              "0x0cac69b3162e2db93af22eb0156a1ecb6d2641e1"
+            ); // a cast by @dwr
+          } catch (error) {
+            expect(NeynarAPIClient.isApiErrorResponse(error)).is.true;
+            if (NeynarAPIClient.isApiErrorResponse(error)) {
+              // repeating the above just for the type inference
+              expect(error.response.status).to.eq(400);
+              expect(error.response.data.code).to.contain(
+                "Could not remove cast"
+              );
+              errorThrownAndParsedCorrectly = true;
+            }
+          }
+          expect(errorThrownAndParsedCorrectly).to.be.true;
+        });
+      });
+
+      describe("cast embeds", function () {
+        it("can publish a cast with URL embeds", async function () {
+          const embedURL = "https://www.farcaster.xyz/";
+          const text = "this is a cast testing URL embed functionality";
+          const publishCastResp = await client.publishCast(
+            signerUuid,
+            text,
+            undefined /* replyTo */,
+            [{ url: embedURL }]
+          );
+          expectDefined(publishCastResp);
+
+          // re-fetch the cast again, since the warpcast API might not return it in response to our POST
+          await sleep(1000);
+          const publishedCast = await client.fetchCast(publishCastResp.hash);
+          expectDefined(publishedCast);
+
+          expectDefined(publishedCast.embeds);
+          const urls = publishedCast.embeds;
+          expect(urls).to.have.length(1);
+          const publishedEmbed: EmbedUrl = urls[0] as EmbedUrl;
+          expect(publishedEmbed.url).to.be.eq(embedURL);
+          await client.deleteCast(signerUuid, publishedCast);
+        });
+
+        it("can publish a cast with image embeds", async function () {
+          const embedURL = "https://i.imgur.com/YPEZebo.png";
+          const text = "this is a cast testing image embed functionality";
+          const publishCastResp = await client.publishCast(
+            signerUuid,
+            text,
+            undefined /* replyTo */,
+            [{ url: embedURL }]
+          );
+          expectDefined(publishCastResp);
+
+          // re-fetch the cast again, since the warpcast API might not return it in response to our POST
+          await sleep(1000);
+          const publishedCast = await client.fetchCast(publishCastResp.hash);
+          expectDefined(publishedCast);
```

Comment:
- Still needed?
- Fix reference to "warpcast"

Reply:
still needed but for a different reason, updated the comment
```
// Fetch the cast, the PostCastResponseCast does not contain the embeds
```